portfolio/src/main/java/com/google/sps/servlets/AuthServlet.java

```diff
@@ -21,28 +21,24 @@
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
+import com.google.gson.Gson;
+import java.util.HashMap;
 
 @WebServlet("/auth")
 public class AuthServlet extends HttpServlet {
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    response.setContentType("text/html");
 
     UserService userService = UserServiceFactory.getUserService();
-    if (userService.isUserLoggedIn()) {
-      String userEmail = userService.getCurrentUser().getEmail();
-      String urlToRedirectToAfterUserLogsOut = "/";
-      String logoutUrl = userService.createLogoutURL(urlToRedirectToAfterUserLogsOut);
+    Gson gson = new Gson();
```

Comment:
I meant you could move this to a class member outside of `doGet`

`public static final Gson gson = new Gson();`

Since you are never modifying Gson it should be thread safe and ok outside of `doGet`